user_data.sh

```diff
@@ -197,4 +197,25 @@ cat > ~/mycron << EOF
 0 0 * * * yum -y update --security
 EOF
 crontab ~/mycron
-rm ~/mycron
\ No newline at end of file
+rm ~/mycron
+
+%{ if onelogin_sync }
+
+cat > /usr/bin/bastion/onelogin_sync.py << 'EOF'
+${onelogin_sync_script}
+EOF
+
+cat > /usr/bin/bastion/onelogin_sync.requirements << 'EOF'
+${onelogin_sync_requirements}
+EOF
+
+chmod 755 /usr/bin/bastion/onelogin_sync.py
+yum -yq install python3 || (apt-get -q update && apt-get -yq install python3-pip)
+pip3 install -r /usr/bin/bastion/onelogin_sync.requirements
+
+crontab -l > ~/mycron
```

Comment:
I like the pattern of keeping non-human owned cron snippets in their own file under /etc/cron.d instead of how this will end up in $CRON_SPOOL_DIR/$USER.  Fewer setup commands and eliminates conflict potential on the shared resource that is the root users crontab.

Reply:
I don't disagree, but I'm following the same pattern as the rest of the module.